src/langcheck/metrics/eval_clients/_anthropic.py

```diff
@@ -0,0 +1,185 @@
+
+        Returns:
+            A list of responses to the prompts. The responses can be None if the
+            evaluation fails.
+        '''
+        config = {"model": "claude-3-haiku-20240307", "max_tokens": 1024}
+        config.update(self._anthropic_args or {})
+        tqdm_description = tqdm_description or 'Intermediate assessments (1/2)'  # NOQA: E501
+        responses = self._call_api(prompts=prompts,
+                                   config=config,
+                                   tqdm_description=tqdm_description)
+        response_texts = [
+            response.content[0].text if response else None
+            for response in responses
+        ]
+
+        return response_texts
+
+    def get_float_score(
+            self,
+            metric_name: str,
+            language: str,
+            unstructured_assessment_result: list[str | None],
+            score_map: dict[str, float],
+            *,
+            tqdm_description: str | None = None) -> list[float | None]:
+        '''The function that transforms the unstructured assessments (i.e. long
+        texts that describe the evaluation results) into scores.
+
+        Args:
+            metric_name : The name of the metric to be used. (e.g. "toxicity")
+            language: The language of the prompts. (e.g. "en")
+            unstructured_assessment_result: The unstructured assessment results
+                for the given assessment prompts.
+            score_map: The mapping from the short assessment results
+                (e.g. "Good") to the scores.
+            tqdm_description: The description to be shown in the tqdm bar.
+
+        Returns:
+            A list of scores for the given prompts. The scores can be None if
+            the evaluation fails.
+        '''
+        if language not in ['en']:
+            raise ValueError(f'Unsupported language: {language}')
+
+        options = list(score_map.keys())
+        get_score_template = get_template(f'{language}/get_score/anthropic.j2')
+        get_score_prompts = [
+            get_score_template.render({
+                'metric': metric_name,
+                'unstructured_assessment': unstructured_assessment,
+                'options': options,
+            }) if unstructured_assessment else None
+            for unstructured_assessment in unstructured_assessment_result
+        ]
+
+        config = {"model": "claude-3-haiku-20240307", "max_tokens": 1024}
+        config.update(self._anthropic_args or {})
+        tqdm_description = tqdm_description or 'Scores (2/2)'
+        responses = self._call_api(prompts=get_score_prompts,
+                                   config=config,
+                                   tqdm_description=tqdm_description)
+        raw_response_texts = [
+            response.content[0].text if response else None
+            for response in responses
+        ]
+
+        def _turn_to_score(response: str | None) -> float | None:
```

Comment:
I think we can use function calling in Anthropic as well:
- https://docs.anthropic.com/claude/docs/tool-use
- https://docs.anthropic.com/claude/docs/tool-use-examples#json-mode

Reply:
As it is mentioned in the API document, the feature is still in public beta, and it is not supported by the `anthropic` Python package. Practically even with Haiku this method seems to work quite well, so I would like to go with this for now. WDYT?

Reply:
> it is not supported by the anthropic Python package

Wait really?😨  Where do you see that?

But yeah it might be better to wait for it to be more widely released, esp. considering this line: "Tool use is not yet available on third-party platforms like Vertex AI or AWS Bedrock, but is coming soon". We can keep your approach for now.

I've also been meaning to look into using https://github.com/jxnl/instructor, and this might be a good opporunity since they seem to have support for Anthropic (caveat: I haven't checked the implementation, so maybe they do something similar to your approach anyway).

Reply:
Here is the official `anthropic-tools` package:
https://github.com/anthropics/anthropic-tools

It says:
> This SDK is Currently in Alpha and meant ONLY AS A RESEARCH PREVIEW. We promise no ongoing support. It is not intended for production use. Production-ready function calling support is coming to the Anthropic API soon.

Reply:
I looked into `instructor`'s implementation, and it seems pretty useful!!

I overlooked that, but `anthropic-sdk-python`'s [README](https://github.com/anthropics/anthropic-sdk-python?tab=readme-ov-file#tool-use-beta) says
> This SDK provides beta support for tool use, aka function calling. More details can be found in [the documentation](https://docs.anthropic.com/claude/docs/tool-use).

and `instructor` uses this when `mode=ANTHROPIC_TOOLS` parameter is given.

Sorry for not checking that carefully, but now I'm okay with either
- using the `tools` API
- using `instructor`
- keeping the current structure

Do you have any preferences?

Reply:
> Here is the official `anthropic-tools` package:
> https://github.com/anthropics/anthropic-tools

Hmm I think this is an older repo or something? It looks like the Beta is in the normal `anthropic` package: https://github.com/anthropics/anthropic-sdk-python?tab=readme-ov-file#tool-use-beta

Reply:
sgtm!!